File: scripts/generate-session.js
const { TelegramClient } = require("telegram");
const { StringSession } = require("telegram/sessions");
const input = require("input");
const fs = require("fs");
const path = require("path");

const API_ID = parseInt(process.env.API_ID || "");
const API_HASH = process.env.API_HASH || "";

if (isNaN(API_ID) || !API_HASH) {
    console.log("Please provide API_ID and API_HASH environment variables or enter them now.");
}

(async () => {
    let apiId = API_ID;
    let apiHash = API_HASH;

    if (!apiId) {
        apiId = parseInt(await input.text("Enter API ID: "));
    }
    if (!apiHash) {
        apiHash = await input.text("Enter API Hash: ");
    }

    const client = new TelegramClient(new StringSession(""), apiId, apiHash, {
        connectionRetries: 5,
    });

    await client.start({
        phoneNumber: async () => await input.text("Phone number (international format): "),
        phoneCode: async () => await input.text("Code: "),
        password: async () => await input.text("2FA Password (if any): "),
        onError: (err) => console.log(err),
    });

    const sessionString = client.session.save();
    console.log("\n✅ Session generated successfully!");
    console.log("\nSESSION STRING (save this to your .env file as TG_SESSION):");
    console.log("---------------------------------------------------");
    console.log(sessionString);
    console.log("---------------------------------------------------");

    // Optional: save to .env if user wants (simple append, careful with duplicates)
    // For now, just printing it is safer.

    await client.disconnect();
    process.exit(0);
})();

File: scripts/generate_session.py
import os
import asyncio
from telethon.sync import TelegramClient
from telethon.sessions import StringSession
from dotenv import load_dotenv

# Load existing .env if present
load_dotenv()

async def generate_session():
    print("🔐 Telethon Session Generator")
    print("----------------------------")

    api_id = os.getenv("API_ID")
    if not api_id:
        api_id = input("Enter API_ID: ").strip()
    
    api_hash = os.getenv("API_HASH")
    if not api_hash:
        api_hash = input("Enter API_HASH: ").strip()

    if not api_id or not api_hash:
        print("❌ Error: API_ID and API_HASH are required.")
        return

    print("\nCheck your Telegram for the login code...")
    
    async with TelegramClient(StringSession(), int(api_id), api_hash) as client:
        session_string = client.session.save()
        print("\n✅ Session String generated successfully!")
        print("\n" + session_string)
        print("\n⚠️  Copy this string and set it as SESSION (or TG_SESSION) in your .env file or GitHub Secrets.")
        print("   Do not share this string with anyone!")

if __name__ == "__main__":
    try:
        asyncio.run(generate_session())
    except ImportError:
        print("❌ Telethon is not installed. Run: pip install telethon")
    except Exception as e:
        print(f"❌ Error: {e}")
